### PR TITLE
Integration Tests for Object Stores and Loggers

### PR DESCRIPTION
**Pull request by ravi-mosaicml**
1. Add remote object store tests
2. Add remote logger tests

Coming along for the ride: some typing fixes

Closes https://mosaicml.atlassian.net/browse/CO-768

### CODE REVIEW

**Review by bandish-shah — COMMENTED** [review]
Is this ready to review?  Started looking and then saw the failed tests, looks like this might need some more work.

**Comment by ravi-mosaicml** [comment]
> Is this ready to review? Started looking and then saw the failed tests, looks like this might need some more work.

Yes, the core logic is here. Fixed (most, if not all) of the failing tests.

File: tests/callbacks/test_callbacks.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def test_multiple_init_and_close(self, cb_cls: Type[Callback], dummy_state: Stat
 
 
 @pytest.mark.parametrize('cb_cls', get_cbs_and_marks(callbacks=True, loggers=True, profilers=True))
-@pytest.mark.parametrize('grad_accum', [1, 2])
+@pytest.mark.parametrize('grad_accum,remote', [(1, False), (2, False), pytest.param(1, True, marks=pytest.mark.remote)])
```
Comment (dskhudia) [inline tests/callbacks/test_callbacks.py:110]: remote is not used in this class?

**Comment by mvpatel2000** [comment]
Discussed offline about `remote`, it's used to toggle `pytest.mark.remote`. Requested comments explaining + changing to `_remote` to indicate its unused variable

**Review by mvpatel2000 — COMMENTED** [review]
No-op review so I can get re-requested for review when ready

File: composer/core/state.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 from typing import TYPE_CHECKING, Any, Callable, Dict, Iterable, Optional, Sequence, Union, cast
 
 import torch
+import torch.nn
```
Comment (mvpatel2000) [inline composer/core/state.py:13]: why is there an import change but no changes in the file?

**Review by mvpatel2000 — APPROVED** [review]
LGTM besides previous comment. Approving to unblock

File: composer/core/state.py (hunk 2 above)
Comment (ravi-mosaicml) [reply composer/core/state.py:13]: Pyright in vscode doesn't like implicit imports of submodules for some reason...

Comment (ravi-mosaicml) [reply composer/core/state.py:13]: (This is coming along for the ride...)

**Comment by ravi-mosaicml** [comment]
> Discussed offline about `remote`, it's used to toggle `pytest.mark.remote`. Requested comments explaining + changing to `_remote` to indicate its unused variable

Added in a comment + renamed the variable.

File: tests/callbacks/test_callbacks.py (hunk 1 above)
Comment (ravi-mosaicml) [reply tests/callbacks/test_callbacks.py]: This is a bit confusing, but it is used to parameterize the test markers. The value of the variable is unused. Added a comment to explain this.

**Pull request merged**